drivers/at24cxxx/at24cxxx.c

```diff
@@ -0,0 +1,366 @@
+/**
+ * @ingroup     drivers_at24cxxx
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the AT24C128 / AT24C256 EEPROM units.
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+
+#define ENABLE_DEBUG  (0)
+#include "debug.h"
+#include "assert.h"
+#include "xtimer.h"
+
+#include "include/at24cxxx_params.h"
+#include "include/at24cxxx_defines.h"
+#include "at24cxxx.h"
+
+/*
+    page 0: 0-63
+    page 1: 64-127
+    page 2: 128-191
+    page 3: 192-255
+    page 4: 256-319
+    page 5: 320-383
+    page 6: 384-447
+    page 7: 448-511
+    page 8: 512-575
+    ...
+ */
+static inline void _position_to_address_words(uint8_t addr[AT24CXXX_ADDR_LEN],
+                                              uint32_t pos)
+{
+    uint16_t a = pos & 0xFFFF;
+
+    addr[0] = (a & 0xFF00) >> 8;
+    addr[1] = a & 0x00FF;
+}
+
+static int _read_byte(const at24cxxx_t *dev, uint32_t pos)
+{
+    uint8_t buffer[AT24CXXX_ADDR_LEN];
+
+    _position_to_address_words(buffer, pos);
+
+    int check;
+
+    //dummy write
+    check = i2c_write_bytes(dev->params.i2c, dev->params.dev_addr, buffer,
+                            sizeof(buffer), I2C_NOSTOP);
+    DEBUG("write_byte check: %d\n", check);
+    if (check < 0) {
+        //check is a negative errno value
+        return check;
+    }
+
+    //current address read
+    check = i2c_read_byte(dev->params.i2c, dev->params.dev_addr, buffer, 0);
+    DEBUG("read_byte check: %d\n", check);
+    if (check < 0) {
+        //check is a negative errno value
+        return check;
+    }
+
+    return *buffer;
+}
+
+static int _read(const at24cxxx_t *dev, uint32_t pos, uint8_t *data, size_t len)
+{
+    uint8_t buffer[AT24CXXX_ADDR_LEN];
+
+    _position_to_address_words(buffer, pos);
+
+    int check;
+
+    //dummy write
+    check = i2c_write_bytes(dev->params.i2c, dev->params.dev_addr, buffer,
+                            sizeof(buffer), I2C_NOSTOP);
+    DEBUG("write check: %d\n", check);
+    if (check < 0) {
+        //check is a negative errno value
+        return check;
+    }
+
+    //sequential read
+    check = i2c_read_bytes(dev->params.i2c, dev->params.dev_addr, data, len, 0);
+    DEBUG("read check: %d\n", check);
+    if (check < 0) {
+        //check is a negative errno value
+        return check;
+    }
+    return len;
+}
+
+static int _write_byte(const at24cxxx_t *dev, uint32_t pos,  uint8_t data)
```

Comment:
This can be replaced by [`i2c_write_reg()`](https://riot-os.org/api/group__drivers__periph__i2c.html#ga02b8328d5e823d972e1e1234965daab1)

Reply:
How about `_write_byte()` calling `i2c_write_reg()`, while it also polls for the EEPROM to respond?